src/am-client.ts

```diff
@@ -0,0 +1,248 @@
+  }
+
+  /**
+   * Returns an OpenAM login URL with the goto query parameter set to the original URL in req.
+   */
+  getLoginUrl(goto?: string, realm = '/'): string {
+    return this.serverUrl + url.format({
+      pathname: '/UI/Login',
+      query: { goto, realm }
+    });
+  }
+
+  /**
+   * Constructs a CDSSO login URL
+   *
+   * @param {string} target Target URL
+   * @param {string} provider ProviderId (app URL)
+   * @return {string}
+   */
+  getCDSSOUrl(target: string, provider: string) {
+    return this.serverUrl + url.format({
+      pathname: '/cdcservlet',
+      query: {
+        TARGET: target,
+        RequestID: shortid.generate(),
+        MajorVersion: 1,
+        MinorVersion: 0,
+        ProviderID: provider,
+        IssueInstant: new Date().toISOString()
+      }
+    });
+  }
+
+  /**
+   * Gets policy decisions from OpenAM for params. params must be a well formatted OpenAM policy request object.
+   * It needs a valid sessionId and cookieName in order to make the request. (The user to whom the session belongs needs
+   * to have the REST calls for policy evaluation privilege in OpenAM.
+   */
+  getPolicyDecision(data: AmPolicyDecisionRequest,
+                    sessionId: string,
+                    cookieName: string,
+                    realm = '/'): Promise<AmPolicyDecision[]> {
+    return Axios
+      .post(this.serverAddress + '/json/policies', data, {
+        headers: {
+          [ cookieName ]: sessionId,
+          host: this.hostname
+        },
+        params: {
+          _action: 'evaluate',
+          realm: realm || '/'
+        }
+      })
+      .then(res => res.data);
+  }
+
+  /**
+   * Sends requestSet to the SessionService. requestSet must be a properly formatted XML document.
+   *
+   * @param {object} requestSet Session service request set
+   * @return {Promise} Session service response
+   */
+  sessionServiceRequest(requestSet: string): Promise<any> {
+    return Axios
+      .post(this.serverAddress + '/sessionservice', requestSet, {
+        headers: {
+          host: this.hostname,
+          'Content-Type': 'text/xml'
+        }
+      })
+      .then(res => res.data);
+  }
+
+  /**
+   * Validates the OAuth2 access_token in the specified realm.
+   *
+   * @param {string} accessToken OAuth2 access_token
+   * @param {string} [realm=/]
+   * @return {Promise} Token info response
+   */
+  validateAccessToken(accessToken: string, realm = '/'): Promise<any> {
+    //noinspection JSValidateTypes
+    return Axios
+      .get(this.serverAddress + '/oauth2/tokeninfo', {
+        headers: {
+          host: this.hostname
+        },
+        params: {
+          access_token: accessToken,
+          realm
+        }
+      })
+      .then(res => res.data);
+  }
+
+  /**
+   * Gets a user's profile (requires an agent or admin session).
+   */
+  getProfile(userId: string, realm: string, sessionId: string, cookieName: string): Promise<any> {
+    //noinspection JSValidateTypes
```

Comment:
What's this comment about? Lint directive?

Reply:
uh... not sure... some old stuff I didn't notice

Reply:
It shows up in a couple other spots as well, I didn't see sense flagging it everywhere I saw it.